src/server/rpc.rs

```diff
@@ -129,28 +129,45 @@ async fn worker_rpc_loop<
         let mut interval = tokio::time::interval(heartbeat_interval);
```

Comment:
Maybe this should be
```rust
let mut interval = tokio::time::interval(std::cmp::min(heartbeat_interval, core.idle_timeout().unwrap_or(heartbeat_interval)));
```
?

Reply:
You are right. I had some internals reasons why not not do it in this way, but when I was writing them here, I realized that it is a bullshit :) [I generally plan to make this interval dynamically configurable, which makes the interval slightly tricky, but this is a problem for our future selves]